docs/fundamentals/k8s.md

```diff
@@ -1,10 +1,138 @@
 # Jina on Kubernetes
 
-Jina is a cloud native framework and therefore run natively on Kubernetes.
+Jina is a cloud native framework and therefore run natively and easily on Kubernetes.
+Deploying a Jina Flow on Kubernetes is actually the recommended way to use Jina in production.
+
+
+A {class}`~jina.Flow` is composed of different microservices called {class}`~jina.Executor`s which natively run in container. This mean that Kubernetes can natively take over the lifetime management of Executors. 
+
+Deploying a {class}`~jina.Flow` on Kubernetes means wrapping these microservices container in the appropriate K8s abstraction (Deployment, StatefulSet, ...) expose them internally via K8s Service and 
+connecting them together by passing the right set of parameters.
+
+## Automatically translate a Flow to Kubernetes concept
+
+
+```{hint}
+Manually building these Kubernetes YAML object is long and cumbersome. Therefore we provide a helper function {function}`~jina.Flow.to_kubernetes_yaml()` that does most of this
+translation work automatically. 
+```
+
+This helper function can be called from:
+* our Python interface to translate a Flow define in Python to k8s yaml files
+* our CLI interface to export a YAML Flow to k8s yaml files
+
+```{seealso}
+More detail in the {ref}` Flow export documentation <kubernetes-export>`
+```
+
+
+
+## Extra Kubernetes options
+
+In general, Jina follows the following principle when it comes to deploying in Kubernetes:
+You, the user, know your use case and requirements the best.
+This means that while Jina generates configurations for you that run out of the box, as a professional user you should
+always see them as just a starting point to get you off the ground.
+
+```{hint}
+The export funciton {function}`~jina.Flow.to_kubernetes_yaml()` is a helper function to get your stared off the ground. **There are meant to be updated and adapted to every use case**
+```
+````{admonition} Matching jina versions
+:class: caution
+If you change the Docker images for {class}`~jina.Executor` and {class}`~jina.Gateway` in your Kubernetes generated file, ensure that al of them are built with the same Jina version to guarantee compatibility.
```

Comment:
```suggestion
````{admonition} Matching Jina versions
:class: caution
If you change the Docker images for {class}`~jina.Executor` and {class}`~jina.Gateway` in your Kubernetes-generated file, ensure that all of them are built with the same Jina version to guarantee compatibility.
```